packages/product/src/loaders/container.ts

```diff
@@ -53,43 +71,123 @@ export default async ({
     productOptionValueService: asClass(ProductOptionValueService).singleton(),
   })
 
+  const defaultProductCollectionRepository =
```

Comment:
thought: Not sure if this is better actually. I would argue this is more error-prone and confusing than having separate files.

If we are to proceed with "hidden repositories", we have to find a way to make them more hidden than this.

Could we create a repository under the hood for each model and use the model as the type for the create and update methods. I am not sure this would suffice, what do you think?

Reply:
Is the types the main reason for having them explicitly defined?

Reply:
Thats what I was trying to explain with all the hidding. I would say that we will loose the DTO typings but at the same type when I look at it we are not really using the DTO typings since in the services the repositories are typed like this `DAL.RepositoryService<TEntity>`. So I am wondering if there is something to do around our typings

Reply:
Maybe we should improve the RepositoryService types to ensure the DTO are applied here and have that as part of the internal module typings, and for the default dto then we don't need to provide them and I can build then automatically

Reply:
maybe we need to huddle around it

Reply:
So "DTOs" for repositories could perhaps just be the models? Or what are you hinting at

Reply:
I ve just pushed a proposal of what I tried to explain, [here](https://github.com/medusajs/medusa/pull/6127/commits/44afcaa7eed0f253fc22ce6af99568aa97f0de3c)

Reply:
meaning, we can still keep the templates on the repository themselves for the future but we can rely on the way I ve done it above